.github/workflows/ci.yml

```diff
@@ -32,83 +63,102 @@ jobs:
   # and for .NET Standard 2.0.
   build:
     if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    needs: setup-build-matrix
     strategy:
-      matrix:
-        os: [ubuntu-18.04, windows-2019, macos-10.15]
+      matrix: ${{ fromJson(needs.setup-build-matrix.outputs.matrix) }}
       fail-fast: false
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Setup .NET Core SDK
-        if: runner.os != 'Windows'
+      - name: Split dotnet variables
+        id: dotnet
+        env:
+          DOTNET: ${{ matrix.dotnet }}
+        shell: bash
+        run: |
+          IFS=':' read -ra dotnet <<< "$DOTNET"
+          if [ "${dotnet[0]}" == "net461" ]
+          then
+            libframework=net461
+          else
+            libframework=netstandard2.0
+          fi
+          echo "::set-output name=framework::${dotnet[0]}"
+          echo "::set-output name=version::${dotnet[1]}"
+          echo "::set-output name=channel::${dotnet[2]}"
+          echo "::set-output name=libframework::$libframework"
+      - name: Setup old .NET SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '2.1.x'
+      - name: Setup old .NET SDK
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.404
+          dotnet-version: '3.1.x'
+      - name: Setup .NET SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '5.0.202'
       - name: Download Consul
+        if: matrix.component == 'Consul'
         shell: bash
         run: |
           cd Consul.Test
           SYSTEM=$(uname | sed 's/MINGW.*/windows/' | tr A-Z a-z)
-          curl -sSL https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_${SYSTEM}_amd64.zip -o consul.zip
+          curl -sSL https://releases.hashicorp.com/consul/${{ matrix.consul }}/consul_${{ matrix.consul }}_${SYSTEM}_amd64.zip -o consul.zip
           unzip consul.zip
           rm consul.zip
-      - name: Build
-        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+      - name: Build library
+        run: dotnet build --framework ${{ steps.dotnet.outputs.libframework }} --configuration=Release -p:Channel=${{ steps.dotnet.outputs.channel }} ${{ matrix.component }}
```

Comment:
Do we actually use mono in the Ci or it is just a remnant of initial attempts ?

Reply:
> Do we actually use mono in the Ci 

Doesn't look like it

Reply:
So I think setting the `Channel` property is not needed.